DuckDuckGo/AppDelegate.swift

```diff
@@ -442,6 +442,19 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 #if NETWORK_PROTECTION
         widgetRefreshModel.refreshVPNWidget()
 #endif
+
+#if SUBSCRIPTION
+        Task {
+            guard let token = AccountManager().accessToken else {
+                return
+            }
+            let result = await SubscriptionService.getSubscriptionDetails(token: token)
+            switch result {
+            case .success: DailyPixel.fire(pixel: .privacyProSubscriptionActive)
```

Comment:
A valid token does not necessarily mean the subscription is active.    It can be inactive or expired and you still can have a valid account.   You'll need to check for the subscription details.  See GetSubscriptionDetailsResponse > isSubscriptionActive

Reply:
Done